nestkernel/connection.h

```diff
@@ -400,7 +402,7 @@ template < typename targetidentifierT >
 SecondaryEvent*
 Connection< targetidentifierT >::get_secondary_event()
 {
-  assert( false );
+  std::abort();
```

Comment:
I still don't quite understand why `abort()` is better than a failing assertion.

Reply:
Yes, `assert(false)` would call `std::abort()`, but as it is a macro which is conditionally removed by the define `NDEBUG` (according to the standard, see https://en.cppreference.com/w/cpp/error/assert) it's not safe (nest might have some non-standard behavior w.r.t. NDEBUG/assert, so I'm not sure if this is the case here).

Reply:
if `abort()` is what should be done here, then we need to explicitly call it here. Especially the production builds (apt, spack, …) all have the build-type "rel" (release) which removes debugging stuff. So the assert resolves to `;` and does not cause nest to `abort()` as is intended.

Reply:
Maybe I'm missing something, but why provide this function at all? There's no dynamic casting going on, as far as I could see. I.e. just remove the function and there would be a build-time error if someone wrongly calls this (as a base class).
Maybe it is about the intend to communicate the idea of an "API" → you could still leave out the implementation, and only a link-time error would occur. Or, if modern C++ is okay,  you could just redefine `Connection` to be a concept that the "derived" Connection classes would have to fulfill.

```
template<typename C>
concept ConnectionLike = requires(C const src, C dst)
{
    C{};
    C{src};
    dst = src;
    
    // a non-const get_secondary_event member function
    dst.get_secondary_event();
};
```

Reply:
@muffgaga Yes, we could use constexpr-evaluation here. But the entire "secondary event" implementation needs a thorough overhaul, so I would not like to spend any significant effort on code improvements in this area now. But I like your suggestion to return a nullptr after the assert (and remove the ``[[noreturn]]``). But then I would really like to wrap up this PR—it is a tiny fix to get to a well-defined state which avoids many compiler warnings and that should not be delayed by possible larger improvements. One can always create a follow-up issue ...

Reply:
Ok, got rid of the other things, and only added a `return nullptr;` (+ a small helpful msg to the assert).